server.js

```diff
@@ -1,23 +1,90 @@
 const express = require("express");
 const cors = require("cors");
-
+const bodyParser = require("body-parser");
+let changeID = 0;
 const app = express();
+app.use(express.json());
 
 app.use(cors());
 
+app.use(bodyParser.urlencoded({ extended: true }));
+
 const welcomeMessage = {
-  id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
+  id: 0
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", function (request, response) {
+  if (messages.length === 0) {
+    response.send("There is no data to view.");
+  } else {
+    response.send(messages);
+  }
+});
+
+app.get("/messages/search", (request, response) => {
+  const term = request.query.term;
+  let filteredMessages = messages.filter((eachMessage) =>
+    eachMessage.text.includes(term)
+  );
+  if (filteredMessages.length === 0) {
```

Comment:
Same as above. The success case should always come foirst & the error case should come last.